File: aoc2020/day5/Main.hs
module Main (main) where

int64 :: String -> Int
int64 = foldl (
    \acc c -> 2 * acc + case c of
      'B' -> 1
      'R' -> 1
      _ -> 0
  ) 0

solve :: String -> (Int, Int)
solve input = (max_id, sum [min_id..max_id] - sum_id)
  where
    ids = map int64 . lines $ input
    min_id = minimum ids
    max_id = maximum ids
    sum_id = sum ids

main :: IO ()
main = interact $ show . solve

File: aoc2020/day5/main.py
def main():
    with open("./input1.txt") as inputs:
        ids = [
            int(line.translate(str.maketrans("BFRL", "1010")), 2)
            for line in inputs
        ]
        min_id, max_id, sum_id = min(ids), max(ids), sum(ids)
        print("Part 1:", max_id)
        print("Part 2:", sum(range(min_id, max_id + 1)) - sum_id)


if __name__ == "__main__":
    main()
